### DISCUSSION TITLE
Is it possible to augment an external type without deriving from it?

### DISCUSSION THREAD

**Discussion by rzvc** [vimeo/psalm#7970] [Q&A] [answered]
Like the title says, I would like to add some properties and methods to an external type that I can't modify directly.

For example:
```php
class Foo
{
}

// <-- How can I specify here that Foo has some extra properties, without touching the main Foo declaration?
```

**Comment by AndrolGenhald** [comment, marked as answer]
You can use [stubs](https://psalm.dev/docs/running_psalm/configuration/#stubs). Class declarations from stubs get merged with the actual declaration, so you don't need to repeat everything, just add what you need.

**Reply by rzvc** [reply]
Awesome, thank you.
